PWGMM/Tasks/analyse-mft-tracks.cxx

```diff
@@ -27,33 +27,99 @@ struct analyseMFTTracks {
+      {"TracksPhiEta_LastClsLastDisk", "; #varphi; #eta; ", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},            //
+      {"TracksPhiEta_LastClsLastDisk_firstFace", "; #varphi; #eta; ", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},  //
+      {"TracksPhiEta_LastClsLastDisk_secondFace", "; #varphi; #eta; ", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}}, //
+
+      {"TracksPhiEtaGen", "; #varphi; #eta; ", {HistType::kTH2F, {{600, 0, 2 * M_PI}, {35, -4.5, -1.}}}},  //
+      {"TracksEtaZvtxGen", "; #eta; Z_{vtx}; ", {HistType::kTH2F, {{35, -4.5, -1.}, {201, -20.1, 20.1}}}}, //
+      {"NtrkZLastCls", "z_{LastCls}; count; ", {HistType::kTH1F, {{26, -79, -66}}}},
       //{"NtrkZvtxGen", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}},            //
-      {"NtrkEtaGen", "#eta; N_{trk}; events", {HistType::kTH1F, {{35, -4.5, -1.}}}}, //
-    }                                                                                //
+      {"NtrkEtaGen", "#eta; N_{trk}; ", {HistType::kTH1F, {{35, -4.5, -1.}}}}, //
+    }                                                                          //
   };
 
   void processRec(o2::aod::Collision const& collision, o2::aod::MFTTracks const& tracks)
   {
 
     auto z = collision.posZ();
     registry.fill(HIST("NtrkZvtx"), tracks.size(), z);
+    TVector3 v;
 
     for (auto& track : tracks) {
+      double zLastCls = track.z();
+      v.SetXYZ(track.x(), track.y(), zLastCls);
+      double etaLastCls = v.Eta();
+      double phiLastCls = v.Phi();
       registry.fill(HIST("TracksPhiEta_in_coll"), track.phi(), track.eta());
       registry.fill(HIST("TracksEtaZvtx"), track.eta(), z);
       registry.fill(HIST("NtrkEta"), track.eta());
+      if (zLastCls < -70) //disk 4
+      {
+        registry.fill(HIST("TracksXLastClsYLastCls"), track.x(), track.y());
+        registry.fill(HIST("TracksPhiEta_LastClsLastDisk"), phiLastCls, etaLastCls);
+
+        if (zLastCls > -77) //layer 8
+        {
+          registry.fill(HIST("TracksXLastClsYLastCls_firstFace"), track.x(), track.y());
+          registry.fill(HIST("TracksPhiEta_LastClsLastDisk_firstFace"), phiLastCls, etaLastCls);
+        }
+      }
+      if (zLastCls < -77) //layer 9
+      {
+        registry.fill(HIST("TracksXLastClsYLastCls_secondFace"), track.x(), track.y());
+        registry.fill(HIST("TracksPhiEta_LastClsLastDisk_secondFace"), phiLastCls, etaLastCls);
+      }
+      registry.fill(HIST("TracksPhiEta_LastCls"), phiLastCls, etaLastCls);
+      registry.fill(HIST("NtrkZLastCls"), track.z());
     }
     icoll++;
   }
   //end of processRec
   PROCESS_SWITCH(analyseMFTTracks, processRec, "Process rec level", true);
 
+  void processTracks(o2::aod::MFTTracks const& tracks)
+  {
+
+    for (auto& track : tracks) {
+      registry.fill(HIST("TracksPhiEta"), track.phi(), track.eta());
+
+      if (track.nClusters() >= 5) {
```

Comment:
Please take a look at `histogramRegistry.cxx` tutorial, lines 111-116, there is a nicer way to fill these histograms.

Reply:
Ok will do